source/install.rst

```diff
@@ -55,164 +54,201 @@ To install latest stable releases, refer to following:
 
-       curl https://raw.githubusercontent.com/E3SM-Project/e3sm_diags/master/conda/e3sm_diags_env_osx.yml
+.. _install_latest:
 
-2. Change ``prefix`` in that file to be your conda prefix. Typically, this will be ``~/miniconda3/envs/e3sm_diags_env``.
+(a) Latest Stable Release
+-------------------------
 
+1. Follow :ref:`"Others/Local" <conda_environment_others>` section for installing Conda.
 
-3. Remove any cached conda packages. This will ensure that you always get the latest packages.
+2. Get the yml file to create an environment.
+
+    ::
+
+        wget https://raw.githubusercontent.com/E3SM-Project/e3sm_diags/master/conda/e3sm_diags_env.yml
 
-   ::
 
-       conda clean --all
+3. Change ``prefix`` in that file to be your conda prefix. Typically, this will be ``~/miniconda3/envs/e3sm_diags_env``.
 
-4. Use conda to create a new environment with ``e3sm_diags`` installed.
+4. Remove any cached conda packages. This will ensure that you always get the latest packages
 
-   Tip: You can change the name of the environment to anything you'd like using ``-n <my_env_name>``.
+    ::
 
-   ::
+        conda clean --all
 
-       conda env create -n e3sm_diags_env -f e3sm_diags_env.yml
-       conda activate e3sm_diags_env
+5. Use conda to create a new environment with E3SM Diags (``e3sm_diags``) included.
 
+    - Tip: Add the flag ``-n <name_of_env>`` to customize the name of the environment
+
+    ::
+
+        conda env create -f e3sm_diags_env.yml
+        conda activate e3sm_diags_env
 
 .. _dev-env:
 
-Development environment
------------------------
+(b) Development Environment
+---------------------------
 
-Unlike the latest stable release (i.e., the user environment), the development environment does not include E3SM Diags,
-as the developer will ``pip install`` their changes to E3SM Diags (see step 6 below).
+Unlike the latest stable release (i.e., the user environment), the development environment does not include E3SM Diags (``e3sm-diags``).
+Instead, the developer will ``pip install .`` to build ``e3sm-diags`` with changes (see step 6 below).
 
-Make sure conda is loaded or installed (see :ref:`above <conda_environment>`).
+Furthermore, the dev environment includes QA tools such as code formatters, linters, and ``pre-commit``.
+**NOTE**: These QA tools are enforced using ``pre-commit`` checks in the CI/CD build, so you must use the dev environment for all contributions
 
-1. Get the developmental yml file to create an environment. For Linux machines:
+1. Follow :ref:`"Others/Local" <conda_environment_others>` section for installing conda.
 
-   ::
+2. Clone your fork and keep in sync with upstream ``master``
 
-       wget https://raw.githubusercontent.com/E3SM-Project/e3sm_diags/master/conda/e3sm_diags_env_dev.yml
+    ::
 
-   For macOS:
+        git clone https://github.com/<your-github-username>/e3sm_diags.git
```

Comment:
Do you have to clone your particular fork? I usually clone the main repo and then set my fork as a remote in addition to `origin` (i.e., the main repo, which it looks like you're calling `upstream`).

Reply:
The common convention is to clone your fork and add the upstream repo to keep your fork `master` in sync with the upstream `master`. There are alternatives like the one you mentioned, but this method is most popular.

- https://docs.github.com/en/github/getting-started-with-github/fork-a-repo#keep-your-fork-synced
- https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow

Reply:
Ah ok, good to know. Thanks